tf-modules/s3-replicator/main.tf

```diff
@@ -3,14 +3,29 @@ provider "aws" {
   profile = var.aws_profile
 }
 
+locals {
+  security_group_ids_set = length(var.security_group_ids) > 0
+}
+
 data "archive_file" "replicator_package" {
   type        = "zip"
-  source_file = "index.js"
-  output_path = "build/replicator.zip"
+  source_file = "${path.module}/index.js"
+  output_path = "${path.module}/build/replicator.zip"
+}
+
+resource "aws_security_group" "s3_replicator_lambda" {
+  count  = local.security_group_ids_set ? 0 : 1
+  vpc_id = var.vpc_id
+  egress {
+    from_port   = 0
+    to_port     = 0
+    protocol    = "-1"
+    cidr_blocks = ["0.0.0.0/0"]
+  }
 }
 
 resource "aws_lambda_function" "s3_replicator" {
-  filename         = "build/replicator.zip"
+  filename         = "${path.module}/build/replicator.zip"
```

Comment:
This could be DRY'd up by saying `filename = archive_file.replicator_package.output_path`